Dockerfile

```diff
@@ -0,0 +1,39 @@
+# The builder image, used to build the virtual environment
+FROM python:3.11-slim-buster as builder
+
+RUN apt-get update && apt-get install -y git
+
+RUN pip install poetry==1.4.2
+
+ENV POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_IN_PROJECT=1 \
+    POETRY_VIRTUALENVS_CREATE=1 \
+    POETRY_CACHE_DIR=/tmp/poetry_cache
+
+ENV HOST=0.0.0.0
+ENV LISTEN_PORT 8000
+EXPOSE 8000
+
+WORKDIR /app
+
+COPY pyproject.toml poetry.lock ./
+COPY ./yt_chat ./yt_chat
+COPY ./README.md ./README.md
+RUN poetry install && rm -rf $POETRY_CACHE_DIR
```

Comment:
The 2-step image building is nice and efficient, but then why do you bother not copying everything (for the first step at least)?
`COPY . .`

Reply:
Also, I guess you don't need to remove POETRY_CACHE_DIR here since it will not end up in the final image anyway.

Reply:
I think it is more efficient for the builder to have limited amount of files, but should not make a big difference

Reply:
POETRY_CACHE_DIR is stored and limit the size of the builder: https://medium.com/@albertazzir/blazing-fast-python-docker-builds-with-poetry-a78a66f5aed0

Reply:
I tested and its 200 vs 900 mb

Reply:
I guess POETRY_CACHE_DIR must be stored inside the .venv folder, which should explain.